### PR TITLE
Generate reports

### PR DESCRIPTION
**Pull request by Talgaon4**
Generate reports

### CODE REVIEW

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 1]
```diff
@@ -318,30 +323,59 @@ public Report fetchInventoryReportByID(Integer reportID) throws Exception {
...
+		// Create a date instance of the month&year but is set to the first day of the month
 		LocalDateTime fromDateTime = date;
 		fromDateTime = fromDateTime.withDayOfMonth(1);
 		fromDateTime = fromDateTime.withHour(0);
 		fromDateTime = fromDateTime.withMinute(0);
 
-		PreparedStatement ps1 = con.getPreparedStatement("SELECT * FROM orders WHERE"
-				+ " date >= '?' AND date < '?' AND area = ?");
+		PreparedStatement ps1 = con.getPreparedStatement("SELECT orderId FROM orders WHERE"
+				+ " date >= '?' AND date < '?' AND location = ?");
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java]: Is there a reason there are quotes around the `?`?

File: EKrut-Server/src/ekrut/server/managers/ServerReportManager.java [hunk 2]
```diff
@@ -1,13 +1,316 @@
...
+		Map<String, Integer> itemsQuantityInOrders = ProcessOrders(orders);
+		
+		// If less than 6 items have been ordered all month then we do not generate an order report
+		if (itemsQuantityInOrders.size() < 6) {
+			return null;
+		}
+		// Save the 5 best sellers items and save the rest of the items into "item" that named "allRest"
+		Map<String, Integer> bestSellers = ProcessItemOrders(itemsQuantityInOrders);
+		// Create a new report object with the generated data		
+		Report report = new Report(null, ReportType.CUSTOMER, date, ekrutLocation, totalOrders, totalOrdersInILS, bestSellers);
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/managers/ServerReportManager.java]: Shouldn't this be `ReportType.ORDER`?

**Comment by almogkh** [comment]
If you want to add 1 to the value of some key in a `Map`, you can do this more easily like so:
```java
map.merge(key, 1, Integer::sum);
```

This will also take care of initializing the value to 1 if the key doesn't exist yet.

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 3]
```diff
@@ -350,8 +387,133 @@ public Object fetchAllMonthlyOrders(LocalDateTime date, String location) {
...
+                                                        "(type,date,location) " +
+                                                        "VALUES(?,?,?)", true);
+		
+		try {
+			p1.setString(1, report.getReportType().toString());
+			p1.setObject(2, report.getDate(), MysqlType.DATETIME);
+			p1.setString(3, report.getEkrutLocation().toString());
+	
+			// Get the new report ID
+			ResultSet rs = p1.getGeneratedKeys();
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java:478]: You need to execute the query first:
```java
int res = p1.executeUpdate();
// check res...
ResultSet rs = p1.getGeneratedKeys();
// ...
```

File: EKrut-Server/src/ekrut/server/db/ReportDAO.java [hunk 4]
```diff
@@ -350,8 +387,133 @@ public Object fetchAllMonthlyOrders(LocalDateTime date, String location) {
...
+	public boolean createOrderReport(Report report) {
+		
+		con.beginTransaction();
+		/* * orders monthly sales table:
+ *   reportID  | totalOrders | totalOrdersInILS |
+ * --------------------------------------------
+ * 	  15678    |    3220    |       18550     |*/
+		
+		
+		PreparedStatement ps1 = con.getPreparedStatement("INSERT INTO orderMonthlySales"
```
Comment (almogkh) [inline EKrut-Server/src/ekrut/server/db/ReportDAO.java:517]: You're not using `ps1` here for anything. Also, the formatting of these lines is a bit broken.

**Comment by almogkh** [comment]
There are additional places in the code where you could use `map.merge()`.

**Comment by almogkh** [comment]
There are a couple of errors in your code that block me from merging this. Fix them and then I can merge.

**Comment by Talgaon4** [comment]
> 

did it :)

**Pull request merged**